### DISCUSSION TITLE
How to use with local models like llama3.1 70b running on ollama

### DISCUSSION THREAD

**Discussion by Gimel12** [gptme/gptme#178] [Q&A] [answered]
Hi, 

I am trying to install it but I want to be able to run it with ollama running as the backend for my LLM on my machine. 

How can we do that?

**Comment by Gimel12** [comment]
I found that there is too many steps for running with local LLms and still not working. 

This is what i did: 

1- Install gpt 
` pip install gptme`

2- Install litellm 
` pip install litellm`

3- Follow the steps on the documentation 
```
ollama pull mistral
ollama serve
litellm --model ollama/mistral
export OPENAI_API_BASE="http://localhost:8000"
```

4- Try to launch gptme on terminal 
`gptme --model local/ollama/llama3.2:1b`

And i keep getting errors: 

  File "/Users/rubenfernandez/.pyenv/versions/3.12.7/lib/python3.12/site-packages/openai/_base_client.py", line 1014, in _request
    return self._retry_request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/rubenfernandez/.pyenv/versions/3.12.7/lib/python3.12/site-packages/openai/_base_client.py", line 1092, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/rubenfernandez/.pyenv/versions/3.12.7/lib/python3.12/site-packages/openai/_base_client.py", line 1014, in _request
    return self._retry_request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/rubenfernandez/.pyenv/versions/3.12.7/lib/python3.12/site-packages/openai/_base_client.py", line 1092, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/rubenfernandez/.pyenv/versions/3.12.7/lib/python3.12/site-packages/openai/_base_client.py", line 1024, in _request
    raise APIConnectionError(request=request) from err
openai.APIConnectionError: Connection error.

**Comment by ProfiGMan** [comment]
The exact same issue here. [Looks like](https://github.com/ErikBjare/gptme/blob/578adcc9228b314a7596711c8021ce6e51eee53e/gptme/models.py#L64) the support for local/ollama is not implemented.

**Comment by ErikBjare** [comment, marked as answer]
Looks like some default ports got changed somewhere since I last tried this.

Here are exact steps I just ran to get it working:

```sh
MODEL=llama3.2:1b
ollama pull $MODEL
ollama serve
OPENAI_API_BASE="http://127.0.0.1:11434" gptme 'hello' -m local/$MODEL
```

I [updated the docs](https://github.com/ErikBjare/gptme/commit/fa59310f9ecc1ff7f613f0956f3d079915f4c486) to reflect this change

**Reply by ErikBjare** [reply]
Apparently ollama directly serves a OpenAI-compatible API now, so litellm no longer needed. Updating answer and docs to reflect this.

**Comment by TimeToBuildBob** [comment]
Late follow-up — this thread is still a top hit for “gptme + ollama”, and the 2024 steps have drifted.

**You do not need LiteLLM.** gptme talks to Ollama’s OpenAI-compatible endpoint directly.

**The 2024 one-liner is missing `/v1`.** Verified on gptme 0.33.0 with a local mock that logged request paths:

```console
$ OPENAI_API_BASE=http://127.0.0.1:PORT gptme -n --no-workspace -m local/llama3.2:1b hello
# mock saw: POST /chat/completions

$ OPENAI_BASE_URL=http://127.0.0.1:PORT/v1 gptme -n --no-workspace -m local/llama3.2:1b hello
# mock saw: POST /v1/chat/completions
```

Ollama’s OpenAI-compat API is the second one. Without `/v1` you never hit it.

**Current recipe** (gptme 0.33.0):

```sh
ollama pull llama3.1:70b
ollama serve   # if it is not already running

OPENAI_BASE_URL=http://127.0.0.1:11434/v1 gptme -m local/llama3.1:70b "hello"
```

To persist instead of exporting every time: `gptme providers add` (name `ollama`, base URL `http://127.0.0.1:11434/v1`). Check discovery with:

```console
$ gptme providers list
🔍 Local auto-discovery
   ⚪ Ollama  http://127.0.0.1:11434/v1
      probe: http://127.0.0.1:11434/v1/models
      not running (connection refused)
```

That probe was run here with Ollama stopped, so I did not do a live generate — the path check above is the load-bearing test.

Missing the env var now fails closed:

```console
$ gptme -n --no-workspace -m local/llama3.2:1b hello
ERROR  'Missing environment variable OPENAI_BASE_URL'
```

`OPENAI_API_BASE` still works as a fallback alias, but use `OPENAI_BASE_URL` and include `/v1`. The `local/ollama/llama3.2:1b` form in the original report is two prefixes; the local provider is `local/<ollama-model-name>`.
